### PR TITLE
Intake state machine changes

### PR DESCRIPTION
**Pull request by Matthew13479**

### CODE REVIEW

File: components/intake.py [hunk 1]
```diff
@@ -12,10 +11,12 @@ class Intake:
     extension_arm_right: wpilib.Solenoid
     infrared: wpilib.AnalogInput
     cube_switch: wpilib.DigitalInput
+    joystick: wpilib.Joystick
 
     def setup(self):
         """This is called after variables are injected by magicbot."""
-        pass
+        self.intake_right.set(self.intake_right.ControlMode.Follower, 2)
```
Comment (auscompgeek) [inline components/intake.py]: What on earth is going on here. Even the tests don't know what ID 2 is.

Use [TalonSRX.follow()](https://robotpy.readthedocs.io/projects/ctre/en/latest/api_talonsrx.html#ctre.talonsrx.TalonSRX.follow), and pass in an actual TalonSRX object.

File: components/intake.py [hunk 2]
```diff
@@ -12,10 +11,12 @@ class Intake:
     extension_arm_right: wpilib.Solenoid
     infrared: wpilib.AnalogInput
     cube_switch: wpilib.DigitalInput
+    joystick: wpilib.Joystick
```
Comment (auscompgeek) [inline components/intake.py]: nonononononononononononononononononononononononononononononono

File: components/intake.py [hunk 3]
```diff
@@ -56,21 +57,18 @@ def infraredDistance(self):
         infrared_voltage = self.infrared.getVoltage()
         """Makes sure that the voltage is above 0.00001"""
         voltage = max(infrared_voltage, 0.00001)
-        distance = 12.84*math.pow(voltage, -0.9824)
-        """This makes sure that the distance is above 4.5 and below 35"""
+        distance = 12.84 * voltage ** -0.9824
+        """This makes sure that the distance is above 4.5cm and below 35cm"""
         self.cube_distance = max(min(distance, 35.0), 4.5)
```
Comment (auscompgeek) [inline components/intake.py]: So, uh, can we please stop reinventing the wheel here? The RobotPy common driver library exists for a reason.

File: components/intake.py [hunk 4]
```diff
@@ -32,7 +33,7 @@ def execute(self):
     def intake_rotate(self, value):
```
Comment (auscompgeek) [inline components/intake.py]: Why is this method name not a verb?

**Review by auscompgeek — CHANGES_REQUESTED** [review]

File: components/intake.py [hunk 5]
```diff
@@ -30,7 +30,7 @@ def execute(self):
         """Run at the end of every control loop iteration."""
         pass
 
-    def intake_rotate(self, value):
+    def rotate_intake(self, value):
```
Comment (auscompgeek) [inline components/intake.py]: Don't I already know that I'm going to rotate the intake if I ask the intake to rotate?

File: components/intake.py [hunk 6]
```diff
@@ -53,22 +53,17 @@ def extension(self, value):
...
-        infrared_voltage = self.infrared.getVoltage()
-        """Makes sure that the voltage is above 0.00001"""
-        voltage = max(infrared_voltage, 0.00001)
-        distance = 12.84 * voltage ** -0.9824
-        """This makes sure that the distance is above 4.5cm and below 35cm"""
-        self.cube_distance = max(min(distance, 35.0), 4.5)
-        print(self.cube_distance, "cm")
+    def infrared_distance(self):
+        self.cube_distance = self.infrared.getDistance()
+        self.cube_distance * 10
```
Comment (auscompgeek) [inline components/intake.py]: what.

File: components/intake.py [hunk 7]
```diff
@@ -30,7 +30,7 @@ def execute(self):
         """Run at the end of every control loop iteration."""
         pass
```
Comment (auscompgeek) [inline components/intake.py]: Why is execute empty?

**Review by auscompgeek — DISMISSED** [review]

**Review by ArthurAllshire — APPROVED** [review]

**Pull request merged**